ui-src/src/cells/holo-vault/components/profile/fieldMapper.tsx

```diff
@@ -53,24 +145,71 @@ class FieldMapper extends React.Component<Props, State> {
   constructor (props: Props) {
     super(props)
     this.state = {
+      suggestions: [],
+      personaAutovalue: props.selectedPersona.name,
+      selectedPersona: props.selectedPersona,
+      fieldAutovalue: '',
       expansionPanelOpen: false,
       mappedPersona: props.selectedPersona,
       mappedField: props.field.name
     }
   }
 
+  static getDerivedStateFromProps (nextProps: Props, prevState: State) {
+    if (nextProps.selectedPersona !== prevState.selectedPersona) {
+      if (nextProps.field.mapping !== undefined) {
+        let mapping = nextProps.field.mapping
+        let filteredField = nextProps.selectedPersona.fields.filter(function (field) {
+          return field.name === mapping.personaFieldName
+        })
+        if (filteredField.length > 0) {
+          mapping.personaHash = nextProps.selectedPersona.hash
+        }
+        let filteredPersonas = nextProps.personas.filter(function (persona: PersonaType) {
+          return mapping.personaHash === persona.hash
+        })
+        if (filteredPersonas.length !== 0) {
+          return {
+            selectedPersona: nextProps.selectedPersona,
+            mappedPersona: filteredPersonas[0],
+            mappedField: nextProps.field.mapping.personaFieldName,
+            personaAutovalue: filteredPersonas[0].name
+          }
+        } else {
+          return {
+            selectedPersona: nextProps.selectedPersona,
+            mappedPersona: nextProps.personas[0],
+            personaAutovalue: nextProps.personas[0].name
+          }
+        }
+      } else {
+        return {
+          selectedPersona: nextProps.selectedPersona,
+          personaAutovalue: nextProps.selectedPersona.name
+        }
+      }
+    }
+    return null
+  }
+
   componentDidMount () {
     if (this.props.field.mapping !== undefined) {
       this.setPersonaAndFieldName(this.props.field)
     }
+    allPersonaSuggestions = []
+    this.props.personas.map((persona: PersonaType) => (
+      allPersonaSuggestions.push(persona.name)
+    ))
   }
 
-  handleMappingChange = (updatedField: ProfileField) => {
-    this.props.handleMappingChange(updatedField)
+  handleMappingChange = (updatedField: ProfileField, value: string) => {
+    console.log('handleMappingChange field')
```

Comment:
Might be worth dumping these console logs before we merge to develop

Reply:
Yep :)